server.py

```diff
@@ -1,36 +1,71 @@
 """Scalable service for generating sequences for SDX (backed by MongoDB)."""
+import datetime
 import settings
-import sys
-import logging
 import logging.handlers
 import os
-from flask import Flask, jsonify
-from pymongo import MongoClient
-
+import psycopg2
+from flask import Flask, jsonify, abort
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import Sequence, exc, event, select
+from sqlalchemy.exc import SQLAlchemyError
+from structlog import wrap_logger
+from structlog.processors import JSONRenderer
+from structlog.stdlib import filter_by_level, add_log_level
+
+__service__ = "sdx-sequence"
 __version__ = "1.3.1"
 
 app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = settings.DB_URL
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db = SQLAlchemy(app)
+
+
+def add_timestamp(_, __, event_dict):
+    event_dict['created'] = datetime.datetime.utcnow().isoformat()
+    return event_dict
+
 
-app.config['MONGODB_URL'] = settings.MONGODB_URL
-logger = settings.logger
+def add_service_and_version(_, __, event_dict):
+    event_dict['service'] = __service__
+    event_dict['version'] = __version__
+    return event_dict
 
-mongo_client = MongoClient(app.config['MONGODB_URL'])
-db = mongo_client.sdx_sequences
+logging.basicConfig(level=settings.LOGGING_LEVEL, format=settings.LOGGING_FORMAT)
+logger = wrap_logger(logging.getLogger(__name__),
+                     processors=[add_log_level,
+                                 filter_by_level,
+                                 add_timestamp,
+                                 add_service_and_version,
+                                 JSONRenderer(indent=1, sort_keys=True)])
+logger.info("START", version=__version__)
 
 
-def get_next_sequence(seq_name):
-    """Get the next sequence number from the database."""
-    next_sequence = db.sequences.find_and_modify(query={'seq_name': seq_name},
-                                                 update={'$inc': {'sequence': 1}},
-                                                 upsert=True, new=True)
+sequence = Sequence("sequence")
+batch_sequence = Sequence("batch_sequence")
+image_sequence = Sequence("image_sequence")
+json_sequence = Sequence("json_sequence")
 
-    return next_sequence['sequence']
+
+def _get_next_sequence(seq):
+    logger.debug("Obtaining next sequence number")
+    try:
+        logger.debug("Getting DB connection")
+        result = db.engine.execute(seq.next_value())
+        logger.debug("Executing get next value on sequence")
+        sequence_no = result.first()[0]
+        logger.debug("Database sequence no is: {}".format(sequence_no))
```

Comment:
Would be preferable to have a separate KV par for the sequence_no - something like 'Retrieved sequence number', sequence_no=sequence_no' possibly